contracts/examples/lottery-esdt/src/lottery.rs

```diff
@@ -266,7 +262,8 @@ pub trait Lottery {
                 self.send().esdt_local_burn(&esdt_token_id, 0, &burn_amount);
             }
 
-            info.prize_pool -= burn_amount;
+            info.prize_pool -= &burn_amount;
+            info.unawarded_amount -= burn_amount;
```

Comment:
Can't comment on unmodified lines, but you should move this whole burn logic to its own function.

And once you do that, you can also de-nest the code by:
```
if burn_percentage == 0 {
     return;
}

// other code here
```